app/pages/sections/additional-services/select/routes.js

```diff
@@ -69,7 +69,12 @@ export const additionalServicesSelectRoutes = (authProvider, addContext, session
     const response = validateAdditionalServicesForm({ data: req.body });
 
     if (response.success) {
-      sessionManager.saveToSession({ req, key: 'selectedAdditionalServiceId', value: req.body.selectAdditionalService });
+      const selectedItemId = req.body.selectAdditionalService;
+      const additionalServices = sessionManager.getFromSession({ req, key: 'additionalServices' });
+      const findCallback = a => a.additionalServiceId === selectedItemId;
+      const selectedItem = additionalServices.find(findCallback);
```

Comment:
Just to tidy up the routes a little. Can you create a little helper function in the controller that does this find.